cfn-resources/cloud-backup-restore-jobs/cmd/resource/resource.go

```diff
@@ -177,8 +179,14 @@ func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 		return *pe, nil
 	}
 
-	clusterName := cast.ToString(currentModel.ClusterName)
-	instanceName := cast.ToString(currentModel.InstanceName)
+	clusterName := ""
+	instanceName := ""
+
+	if *currentModel.InstanceType == clusterInstanceType {
+		clusterName = cast.ToString(currentModel.InstanceName)
```

Comment:
knitpit: I would have take advantage of having only one cluster/serverless id in the model and will have the code in that way instead of going back to have 2 vars